internal/controller/k0smotron.io/k0smotroncluster_etcd.go

```diff
@@ -299,6 +299,16 @@ func (r *ClusterReconciler) initialCluster(kmc *km.Cluster, replicas int32) stri
 
 func (r *ClusterReconciler) generateEtcdInitContainers(kmc *km.Cluster) []v1.Container {
 	return []v1.Container{
+		{
+			Name:            "dns-check",
+			Image:           kmc.Spec.GetImage(),
+			ImagePullPolicy: v1.PullIfNotPresent,
+			Command:         []string{"/bin/bash", "-c"},
+			Args:            []string{"getent ahostsv4 ${HOSTNAME}.${SVC_NAME}." + kmc.Namespace + ".svc"},
+			Env: []v1.EnvVar{
+				{Name: "SVC_NAME", Value: kmc.GetEtcdServiceName()},
+			},
+		},
```

Comment:
What's the purpose of this? I get that it checks the etcd svc can resolve, but why do we need it?

Reply:
The check ensures the node is accessible from the other nodes during the join process.